src/tests/ftest/process_cores.py

```diff
@@ -0,0 +1,454 @@
+                        status = False
+                        continue
+
+                # List the local core file
+                try:
+                    display(args, run_command(['ls', '-l', local_core_file]), 0)
+                except RuntimeError as error:
+                    display(args, "Unexpected error - no local core file found", 0)
+                    status = False
+                    continue
+
+                # Generate a stack trace
+                try:
+                    generate_stacktrace(local_core_file, args)
+                except RuntimeError as error:
+                    display(args, error, 0)
+                    status = False
+
+    # Remove any core files from the remote hosts
+    if args.remove:
+        display(args, "Removing core files from {}".format(args.hosts), 0)
+        commands = [
+            "for file in {}/core.*".format(args.source),
+            "do rm $file",
+            "done",
+        ]
+        if not spawn_commands(list(args.hosts), "; ".join(commands), timeout=1800):
+            display(args, "  Errors removing core files", 0)
+            status = False
+
+    return status
+
+
+def generate_stacktrace(core_file, args):
+    """Generate a stacktrace from the core file.
+
+    Args:
+        core_file (str): full path and file name of the core file
+        args (argparse.Namespace): command line arguments for this program
+
+    Raises:
+        RuntimeError: if there was an error generating a stacktrace
+
+    """
+    core_file_name = os.path.basename(core_file)
+    stack_trace_file = os.path.join(args.destination, ".".join([core_file_name, "stacktrace"]))
+
+    display("  Generating {} from {}".format(stack_trace_file, core_file), 0)
+    try:
+        command = ["gdb", "-c", core_file, "-ex", "info proc exe", "-ex", "quit"]
+        gdb_output = run_command(command)
+        last_line = gdb_output.splitlines()[-1]
+        cmd = last_line[7:-1]
+        # assume there are no arguments on cmd
+        find_char = "'"
+        if cmd.find(" ") > -1:
+            # there are arguments on cmd
+            find_char = " "
+        exe_name = cmd[0:cmd.find(find_char)]
+    except RuntimeError:
+        exe_name = None
+
+    if exe_name:
+        cmd = [
+            "gdb", "-cd={}".format(args.destination),
+            "-ex", "set pagination off",
+            "-ex", "thread apply all bt full",
+            "-ex", "detach",
+            "-ex", "quit",
+            exe_name, core_file_name
+        ]
+
+        try:
+            with open(stack_trace_file, "w") as stack_trace:
+                stack_trace.writelines(get_output(cmd, check=False))
+        except IOError as error:
+            raise RuntimeError("Error writing {}: {}".format(stack_trace_file, error))
+        except RuntimeError as error:
+            raise RuntimeError("Error creating {}: {}".format(stack_trace_file, error))
```

Comment:
(pylint-raise-missing-from) Consider explicitly re-raising using the 'from' keyword

Reply:
Fixed